src/components/AppLoading.tsx

```diff
@@ -25,6 +28,41 @@ export const AppLoading: FC<PropsWithChildren> = ({ children }) => {
 
   const isLoading = !isLoadingComplete || isSignedIn === null || !isLayoutReady
 
+  const currentVersion = Application.nativeApplicationVersion || 'unknown'
+  const os = Platform.OS
+
+  const { mutate: checkForUpdate } = useSystemControllerCheckForAppUpdate()
+
```

Comment:
It can be custom hook, which check the update is necessary.

Reply:
Totally agree